src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::env;
+use uuid::Uuid;
+use openssl::error::ErrorStack;
+use openssl::ssl::{SslConnector, SslMethod};
+use postgres::{error::SqlState, Client, Error, Transaction};
+use postgres_openssl::MakeTlsConnector;
+
+// BEGIN ssl_config
+fn ssl_config() -> Result<MakeTlsConnector, ErrorStack> {
+    let builder = SslConnector::builder(SslMethod::tls())?;
+    Ok(MakeTlsConnector::new(builder.build()))
+}
+// END ssl_config
+
+/// Runs op inside a transaction and retries it as needed.
+/// On non-retryable failures, the transaction is aborted and
+/// rolled back; on success, the transaction is committed.
+// BEGIN execute_txn
+fn execute_txn<T, F>(client: &mut Client, op: F) -> Result<T, Error>
+where
+    F: Fn(&mut Transaction) -> Result<T, Error>,
+{
+    let mut txn = client.transaction()?;
+    loop {
+        let mut sp = txn.savepoint("cockroach_restart")?;
```

Comment:
I actually don't even know enough CockroachDB to understand why a save point is needed here! Could you add a comment explaining it? 😁

Reply:
Ok. It's a special retry savepoint.

Reply:
Done.